config/v1/types_infrastructure.go

```diff
@@ -435,6 +435,11 @@ type AWSPlatformStatus struct {
 	// +kubebuilder:validation:MaxItems=25
 	// +optional
 	ResourceTags []AWSResourceTag `json:"resourceTags,omitempty"`
+
+	// AWSClusterDNSConfig contains all the DNS config required to configure a custom DNS solution.
+	// This field will be set only when the userConfiguredDNS feature is enabled.
+	// +optional
+	AWSClusterDNSConfig *ClusterDNSConfig `json:"awsClusterDNSConfig,omitempty"`
```

Comment:
Given this is already in an AWS struct, do you think we need AWS in the name of this struct?

I'd like to expand the `userConfiguredDNS` feature comment, I'd like to make it obvious that this is an installer feature that is at install time, else it could lead users to think, ok, how do I go and turn that on